File: lib/day-01/calories-counting.rs
use std::fs;

fn main() {
    let contents =
        fs::read_to_string("./lib/day-01/inp.txt").expect("Should have been able to read the file");

    let mut parsed_data: Vec<i32> = contents
        .split("\n\n")
        .map(|itms| {
            itms.split("\n")
                .map(|itm| itm.parse::<i32>().unwrap())
                .sum()
        })
        .collect();
    parsed_data.sort();
    parsed_data.reverse();

    println!("{:?}", parsed_data[0]);
    println!("{:?}", parsed_data[0] + parsed_data[1] + parsed_data[2]);
}

File: python/day-01/calories-counting.py
inp = open('./lib/day-01/inp.txt').read()

elves = inp.split("\n\n")

max_calories = []

for elf in elves:
    snacks = elf.split('\n')
    total_calories = 0
    for snack in snacks:
        total_calories = total_calories + int(snack)
    max_calories.append(total_calories)

max_calories.sort(reverse=True)

print(max_calories[0])
print(max_calories[0] + max_calories[1] + max_calories[2])
